modules/nextflow/src/main/groovy/nextflow/scm/GithubRepositoryProvider.groovy

```diff
@@ -76,13 +76,7 @@ class GithubRepositoryProvider extends RepositoryProvider {
     /** {@inheritDoc} */
     @Override
     String getCloneUrl() {
-        Map response = invokeAndParseResponse( getEndpointUrl() )
-
-        def result = response.get('clone_url')
-        if( !result )
-            throw new IllegalStateException("Missing clone URL for: $project")
-
-        return result
+        return "${config.server}/${project}.git"
```

Comment:
Considering the method `invokeAndParseResponse` is annotated with `@memoized` it should be invoked only once. 

https://github.com/nextflow-io/nextflow/blob/4a6b54aa12a9f5e9bdb83afe26e233440c4bc383/modules/nextflow/src/main/groovy/nextflow/scm/RepositoryProvider.groovy#L279-L280

Not sure why this is happening.

Reply:
It only happens when there are many concurrent nextflow runs. 

It is being performed in AWS Batch, so it doesn't have the opportunity to use the cache.

Reply:
Understand. However I believe this would solve much because in case other API calls will be made to access the config file, cloning the repo, etc. 

Is this not happening because the GitHub API token is not specified?

Reply:
Ah right ok, didn't realize the config file was read that way.

Cloning the repo luckily doesn't use the rate quota.

I would rather not have users configure GitHub creds, but maybe I can find another solution for this.